packages/firestore/test/unit/specs/spec_test_runner.ts

```diff
@@ -323,9 +323,28 @@ class EventAggregator implements Observer<ViewSnapshot> {
   }
 }
 
-interface OutstandingWrite {
-  mutations: Mutation[];
-  userCallback: Deferred<void>;
+/**
+ * FIFO queue that tracks all outstanding mutations for a single test run.
+ * As these mutations are shared among the set of active clients, any client can
+ * add or retrieve mutations.
+ */
+// PORTING NOTE: Multi-tab only.
+class OutstandingMutationTracker {
+  private mutations: Mutation[][] = [];
+
+  push(mutation: Mutation[]): void {
+    this.mutations.push(mutation);
+  }
+
+  peek(): Mutation[] {
+    assert(this.mutations.length > 0, 'No pending mutations');
+    return this.mutations[0];
+  }
+
+  poll(): Mutation[] {
```

Comment:
Is poll() a standard name or a Java-ism?  I'd lean toward shift() to be more JS-idiomatic.

Reply:
https://stackoverflow.com/questions/1590247/how-do-you-implement-a-stack-and-a-queue-in-javascript has 939 upvotes (that must mean it is the standard) and uses shift.